pkgs/stdenv/generic/builder.sh

```diff
@@ -18,6 +18,6 @@ mkdir $out
 # Allow the user to install stdenv using nix-env and get the packages
 # in stdenv.
 mkdir $out/nix-support
-if [ "$propagatedUserEnvPkgs" ]; then
+if [[ -n "${propagatedUserEnvPkgs:-}" ]]; then
```

Comment:
Unless the `-u` setting has changed, I would keep this as "$propgatedUserEnvPkgs"

Reply:
this change is more for styling consistency. This is how everywhere else non empty variables are being checked.
I don't know the edge case that this is suppose to cover. I would definitely want more information on why it's everywhere like this from one of the person who implemented it.